### PR TITLE
HPCC-16551 Ensure timing calls surround an external function call

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

### CODE REVIEW

**Comment by ghalliday** [comment]
@shamser please review

File: ecl/hqlcpp/hqlcpp.cpp [hunk 1]
```diff
@@ -3028,7 +3028,10 @@ void HqlCppTranslator::buildExpr(BuildCtx & ctx, IHqlExpression * expr, CHqlBoun
         return;
     case no_externalcall:
     case no_call:
-        doBuildExprCall(ctx, expr, tgt);
+        if (isTimed(expr))
+            buildTempExpr(ctx, expr, tgt);
+        else
+            doBuildExprCall(ctx, expr, tgt);
```
Comment (shamser) [inline ecl/hqlcpp/hqlcpp.cpp]: I assume it's okay that op=>no_call will never time call?

**Review by shamser — APPROVED** [review]
@ghalliday Back to you

**Comment by ghalliday** [comment]
@shamser code updated to support time on embedded, and test case added.  (Attilla is checking why it failed, since it ran fine for me.)

**Comment by shamser** [comment]
@ghalliday Looks good.

**Comment by ghalliday** [comment]
@richardkchapman

**Pull request merged**